plutus-contract/src/Language/Plutus/Contract/Trace.hs

```diff
@@ -91,9 +96,16 @@ import           Wallet.API                                      (WalletAPIError
 import           Wallet.Emulator                                 (EmulatorAction, EmulatorState, MonadEmulator, Wallet)
 import qualified Wallet.Emulator                                 as EM
 
+-- | Error produced while running a trace. Either a contract-specific
```

Comment:
The problem with this approach is that I'll want to throw more kinds of errors from contracts than just `ContractError`s (e.g. the state machine contract lib will throw some, and your overall error type needs to embed them all). I guess this is okay, since we have a concrete type so I can probably use the `with*Error` functions to get what I want.

Reply:
For the state machine lib I think it's ok to have a `AsStateMachineError` constraint, I just wanted to get rid of `AsAssertionError` because it has nothing to do with the contract, it's just a by-product of running the traces in the emulator

Reply:
Yeah, I think it will be fine.